File: Java/1576.java
class Solution {
    public String modifyString(String s) {
        int val;
        for(int i = 0; i < s.length(); i++){
            if(s.charAt(i) == '?'){
                try{
                    val = 97;
                    while(true){
                        s = s.substring(0,i) + (char)val + s.substring(i+1);
                        if(s.charAt(i) != s.charAt(i-1) && s.charAt(i) != s.charAt(i+1)) break;
                        val++;
                    }
                }
                catch(Exception e){
                    if(s.length()>1){
                        if(i==0){
                            if(s.charAt(i+1) != '?') s = (char)((s.charAt(i+1)-96)%26 + 97) + s.substring(i+1);
                            else s = "a" + s.substring(i+1);
                        }
                        else s = s.substring(0,i) + (char)((s.charAt(i-1)-96)%26 + 97);
                    }
                    else s = "a";
                }
            }
        }
        return s;
    }
}

File: Python/1576.py
class Solution(object):
    def modifyString(self, s):
        s = list(s)
        for i in range(len(s)):
            if s[i] == "?":
                try:
                    val = 97
                    while True:
                        s[i] = chr(val)
                        val += 1
                        if s[i-1] != s[i] and s[i+1] != s[i]:
                            break
                except:
                    if len(s) > 1:
                        if i == 0:
                            s[i] = chr((ord(s[i+1])+1 - 97)%26 + 97)
                        else:
                            s[i] = chr((ord(s[i-1])+1 - 97)%26 + 97)
                    else:
                        s[i] = "a"
        return "".join(s)